recipes/msys2/all/conanfile.py

```diff
@@ -30,27 +29,14 @@ def source(self):
         # source files downloaded will be different based on architecture or OS
         pass
 
-    def _download(self, url, sha256):
-        from six.moves.urllib.parse import urlparse
-        filename = os.path.basename(urlparse(url).path)
-        tools.download(url, filename)
-        tools.check_sha256(filename, sha256)
-        return filename
 
     @property
     def _msys_dir(self):
         return "msys64" if self.settings.arch_build == "x86_64" else "msys32"
 
     def build(self):
         arch = 0 if self.settings.arch_build == "x86" else 1  # index in the sources list
-        url = self.conan_data["sources"][self.version][arch]["url"]
-        sha256 = self.conan_data["sources"][self.version][arch]["sha256"]
-        filename = self._download(**self.conan_data["sources"][self.version][arch])
-        tar_name = filename.replace(".xz", "")
-        self.run("7z.exe x {0}".format(filename))
-        self.run("7z.exe x {0}".format(tar_name))
-        os.unlink(filename)
-        os.unlink(tar_name)
+        tools.get(**self.conan_data["sources"][self.version][arch])
```

Comment:
This breaks compatible with Python < 3.3 and also with all Python interpreters with disabled LZMA support.

Reply:
OK... It was too easy...